### ISSUE TITLE
heasarc query_region parse problem

### ISSUE THREAD

**Issue by caseyjlaw** [astropy/astroquery#1327]
Seems that the latest release introduces an issue with the heasarc region query. I am using python 3.6 and astropy 2.0.11.

This works well in astroquery v 0.3.8:
```
>>> heasarc = Heasarc() 
>>> mission = 'first' 
>>> coords = SkyCoord('12h29m06.70s +32d03m08.7s', frame='icrs') 
>>> table = heasarc.query_region(coords, mission=mission, radius='1 degree')    
>>> len(table)
272
```

But for astroquery v 0.3.9:
```
>>> heasarc = Heasarc() 
>>> mission = 'first' 
>>> coords = SkyCoord('12h29m06.70s +32d03m08.7s', frame='icrs') 
>>> table = heasarc.query_region(coords, mission=mission, radius='1 degree')                                                                                                                   
---------------------------------------------------------------------------
InvalidQueryError                         Traceback (most recent call last)
<ipython-input-5-92e6d8c3137b> in <module>
      2 mission = 'first'
      3 coords = SkyCoord('12h29m06.70s +32d03m08.7s', frame='icrs')
----> 4 table = heasarc.query_region(coords, mission=mission, radius='1 degree')

~/anaconda/envs/py36/lib/python3.6/site-packages/astroquery/utils/class_or_instance.py in f(*args, **kwds)
     23         def f(*args, **kwds):
     24             if obj is not None:
---> 25                 return self.fn(obj, *args, **kwds)
     26             else:
     27                 return self.fn(cls, *args, **kwds)

~/anaconda/envs/py36/lib/python3.6/site-packages/astroquery/utils/process_asyncs.py in newmethod(self, *args, **kwargs)
     27             if kwargs.get('get_query_payload') or kwargs.get('field_help'):
     28                 return response
---> 29             result = self._parse_result(response, verbose=verbose)
     30             self.table = result
     31             return result

~/anaconda/envs/py36/lib/python3.6/site-packages/astroquery/heasarc/core.py in _parse_result(self, response, verbose)
    216         elif "ERROR" in response.text:
    217             raise InvalidQueryError("unspecified error from HEASARC database. "
--> 218                                     "\nCheck error message: \n{!s}".format(response.text))
    219 
    220         try:

InvalidQueryError: unspecified error from HEASARC database. 
Check error message: 
SIMPLE  =                    T / BROWSE (FITSDISPLAY V0.1)                      BITPIX  =                    8                                                  NAXIS   =                    0 / DUMMY HDU                                      EXTEND  =                    T / EXTENSIONS ALLOWED                             END                                                                   
...
(FITS header and table with query results)
```

Note that the error seems to return a FITS table with the same results as in a successful query (in astroquery 0.3.8). Seems that there is an issue in parsing that result to return a proper table.

**Comment by keflavich** [comment]
I don't see this failure with the latest astropy + astroquery:
`0.3.10.dev5159 3.2.dev23515`
Can you try with a more recent version of astropy and see if it persists?

**Comment by caseyjlaw** [comment]
I do see this error for the lastest versions on pip (astropy 3.1.1 and astroquery 0.3.9).
I've rolled back to astroquery 0.3.8 and it works fine.

**Comment by keflavich** [comment]
Could you try with master?  You should be able to install with `pip install --pre`, or failing that, by following the source instructions here: https://astroquery.readthedocs.io/en/latest/#installation

I... now can reproduce this issue, though.  Hm.

**Comment by keflavich** [comment]
It appears that the issue is that we're now raising an error where before we weren't, even though maybe an error was occurring?  This is a bug for sure...

**Comment by bsipocz** [comment]
Fixed by #1338
